// betfairy/index.js
'use strict';
var auth = require('./lib/auth');
var utils = require('./lib/utils');
var betting = require('./lib/betting');
var account = require('./lib/account');

function Betfairy(applicationKey, username, password, debug) {
    if (!(this instanceof Betfairy)) {
        return new Betfairy(applicationKey, username, password, debug);
    }

    this.config = {};

    this.set('account', 'https://api.betfair.com/exchange/account/rest/v1.0/');
    this.set('betting', 'https://api.betfair.com/exchange/betting/rest/v1.0/');
    this.set('auth', 'https://identitysso.betfair.com/api/');
    this.set('applicationKey', applicationKey);
    this.set('username', username);
    this.set('password', password);
    this.set('debug', debug);
    
    this.log('(constructor) Betfairy instance created');
}

Betfairy.prototype.log = utils.log;
Betfairy.prototype.set = utils.set;
Betfairy.prototype.get = utils.get;
Betfairy.prototype.setupRequestOptions = utils.setupRequestOptions;
Betfairy.prototype.api = utils.api;
Betfairy.prototype.betting = betting;
Betfairy.prototype.account = account;
Betfairy.prototype.login = auth.login;
Betfairy.prototype.logout = auth.logout;
Betfairy.prototype.keepAlive = auth.keepAlive;

module.exports = Betfairy;
